src/jrd/SysFunction.cpp

```diff
@@ -2340,26 +2340,20 @@ dsc* evlCharToUuid(thread_db* tdbb, const SysFunction* function, const NestValue
 		}
 	}
 
-	// convert to binary representation
-	char buffer[GUID_BUFF_SIZE];
-	buffer[0] = '{';
-	buffer[37] = '}';
-	buffer[38] = '\0';
-	memcpy(buffer + 1, data, GUID_BODY_SIZE);
-
-	USHORT bytes[16];
-	sscanf(buffer, GUID_NEW_FORMAT,
+#if defined(_MSC_VER) && _MSC_VER < 1900
+#pragma message("Support of hh size modifier is not sure")
+#endif
+
+	UCHAR bytes[16];
+	sscanf(reinterpret_cast<const char*>(data),
+		"%02hhX%02hhX%02hhX%02hhX-%02hhX%02hhX-%02hhX%02hhX-%02hhX%02hhX-%02hhX%02hhX%02hhX%02hhX%02hhX%02hhX",
```

Comment:
I'd suggest putting this "magic constant" format into Guid.h. Something like GUID_NETWORK_FORMAT and GUID_NATIVE_FORMAT maybe?

Reply:
This is the only place where this format is used so I prefer to keep both places (format and parameters) on single screen for better readability. With the constant defined somewhere else it would be easy to mismatch them.